dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/README.md

```diff
@@ -764,4 +764,80 @@ dotnet_diagnostic.ACL1014.allowed_words = B2C,365
 ```
 Note the following:
 - These overrides are case-insensitive. For example. an identifier with 'b2c' in its name would not violate this rule.
-- These overrides can contain letters. For example, an identifier with '2' in its name would violate this rule.
\ No newline at end of file
+- These overrides can contain letters. For example, an identifier with '2' in its name would violate this rule.
+
+## ACL1015 - Controller action has ProducesResponseType attribute when return type is TypedResults
+
+<table>
+<tr>
+    <td>Category:</td>
+    <td>Maintainability</td>
+</tr>
+<tr>
+    <td>Audacia coding standard:</td>
+    <td>N/A</td>
+</tr>
+</table>
+
+ACL1015 checks if a controller action has `ProducesResponseType` attribute when the return type is TypedResults and will produce a warning if that is the case.
+This applies to controller actions defined by either an Http attribute (eg. `HttpGet`) or by the parent class inheriting the `ControllerBase` class.
+
+Code with diagnostic:
+
+```csharp
+[HttpGet]
+[ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
+public public Results<NotFound, Ok<string>> Get()
+{
+    var result = GetResult();
+    result == null ? TypedResults.NotFound() : TypedResults.Ok(result);
+}
+```
+
+Code without diagnostic:
+
+```csharp
+[HttpGet]
+public public Results<NotFound, Ok<string>> Get()
+{
+    var result = GetResult();
+    result == null ? TypedResults.NotFound() : TypedResults.Ok(result);
+}
+```
+
+## ACL1016 - Controller actions should return TypedResults instead of IActionResults
+
+<table>
+<tr>
+    <td>Category:</td>
+    <td>Maintainability</td>
+</tr>
+<tr>
+    <td>Audacia coding standard:</td>
+    <td>N/A</td>
+</tr>
+</table>
+
+ACL1016 checks if a controller action return type is TypedResults instead of IActionResult and will produce a warning if not.
+This applies to controller actions defined by either an Http attribute (eg. `HttpGet`) or by the parent class inheriting the `ControllerBase` class.
+
+Code with diagnostic:
+
+```csharp
+[HttpGet]
+public public IActionResult Get()
+{
+    return Ok('hello');
+}
+```
+
+Code without diagnostic:
+
+```csharp
+[HttpGet]
+public public Results<NotFound, Ok<string>> Get()
+{
+    var result = GetResult();
+    result == null ? TypedResults.NotFound() : TypedResults.Ok(result);
```

Comment:
can you copy/past these code snippets into VS or something to make sure they compile? Would save my trying to spot things like this missing a `return`

Reply:
will double check all the new snippets